app/src/main/cpp/skyline/applet/swkbd/software_keyboard_config.h

```diff
@@ -0,0 +1,187 @@
+     * @url https://switchbrew.org/wiki/Software_Keyboard#InitialCursorPos
+     */
+    enum class InitialCursorPos : u32 {
+        First = 0x0,
+        Last = 0x1,
+    };
+
+    /**
+     * @url https://switchbrew.org/wiki/Software_Keyboard#PasswordMode
+     */
+    enum class PasswordMode : u32 {
+        Show = 0x0,
+        Hide = 0x1, //!< Hides any inputted text to prevent a password from being leaked
+    };
+
+    /**
+     * @url https://switchbrew.org/wiki/Software_Keyboard#InputFormMode
+     * @note Only applies when 1<= textMaxLength <= 32, otherwise Multiline is used
+     */
+    enum class InputFormMode : u32 {
+        OneLine = 0x0,
+        MultiLine = 0x1,
+        Separate = 0x2, //!<Used with separateTextPos
+    };
+
+    /**
+     * @brief Specifies the language of custom dictionary entries
+     * @url https://switchbrew.org/wiki/Software_Keyboard#DictionaryLang
+     */
+    enum class DictionaryLang : u16 {
+        Japanese = 0x00,
+        AmericanEnglish = 0x01,
+        CanadianFrench = 0x02,
+        LatinAmericanSpanish = 0x03,
+        Reserved1 = 0x04,
+        BritishEnglish = 0x05,
+        French = 0x06,
+        German = 0x07,
+        Spanish = 0x08,
+        Italian = 0x09,
+        Dutch = 0x0A,
+        Portuguese = 0x0B,
+        Russian = 0x0C,
+        Reserved2 = 0x0D,
+        SimplifiedChinesePinyin = 0x0E,
+        TraditionalChineseCangjie = 0x0F,
+        TraditionalChineseSimplifiedCangjie = 0x10,
+        TraditionalChineseZhuyin = 0x11,
+        Korean = 0x12,
+    };
+
+    /**
+     * @brief Info about a custom dictionary entry
+     * @url https://switchbrew.org/wiki/Software_Keyboard#DictionaryInfo
+     */
+    struct DictionaryInfo {
+        u32 offset;
+        u16 size;
+        DictionaryLang language;
+    };
+
+    /**
+     * @brief The keyboard config that's common across all versions
+     * @url https://switchbrew.org/wiki/Software_Keyboard#KeyboardConfig
+     */
+    struct CommonKeyboardConfig {
+        KeyboardMode keyboardMode;
+        std::array<char16_t, 0x9> okText;
+        char16_t leftOptionalSymbolKey;
+        char16_t rightOptionalSymbolKey;
+        bool isPredictionEnabled;
+        u8 _pad0_[0x1];
+        InvalidCharFlags invalidCharFlags;
+        InitialCursorPos initialCursorPos;
+        std::array<char16_t, 0x41> headerText;
+        std::array<char16_t, 0x81> subText;
+        std::array<char16_t, 0x101> guideText;
+        u8 _pad2_[0x2];
+        u32 textMaxLength;
+        u32 textMinLength;
+        PasswordMode passwordMode;
+        InputFormMode inputFormMode;
+        bool isUseNewLine;
+        bool isUseUtf8;
+        bool isUseBlurBackground;
+        u8 _pad3_[0x1];
+        u32 initialStringOffset;
+        u32 initialStringLength;
+        u32 userDictionaryOffset;
+        u32 userDictionaryNum;
+        bool isUseTextCheck;
+        u8 reserved0[0x3];
```

Comment:
There is no other `reserved` member, this should just be `reserved` not `reserved0`.